app/models/doi_search.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class DoiSearch
+  attr_reader :doi
+
+  def initialize(doi:)
+    @doi = Doi.new(doi)
+  end
+
+  # @returns an array of uuids for the matching resources
+  def results
+    return [] unless doi.valid?
+
+    doi_query = escape(doi.to_s)
+
+    solr_response = Blacklight.default_index.search(
+      q: %(all_dois_ssim:"#{doi_query}"),
+      fl: ['id'],
+      rows: max_documents
+    )
+
+    solr_response
+      .response
+      .fetch('docs', [])
+      .map { |d| d['id'] }
```

Comment:
@awead Is blacklight guaranteed to have a `.docs` that will always return an array?

Reply:
@awead The reason why I did it this way was to access the hash directly, without having it mapped to a `SolrDocument`. Since we are simply grabbing a single key verbatim, I didn't think it was worth the performance hit of  mapping those to a SolrDocument

```ruby
> response = Blacklight.default_index.search(q: 'all_dois_ssim:*', rows: 1_000, fl: ['id']) # => Blacklight::Solr::Response

> response.docs.first.class => SolrDocument

> response.response.fetch('docs').first.class => Hash
```

Reply:
This is way all the other searches work, so by running it through Blacklight::Solr::Response, etc. you're guaranteed to be interfacing with Solr the same way. The hash response from Solr could change, too, and RSolr/Blacklight would take care of all that.

Reply:
OK, I'll update that method to your version 😎